src/System.Security.Cryptography.Cng/src/System/Security/Cryptography/CngKey.Import.cs

```diff
@@ -35,9 +47,70 @@ public static CngKey Import(byte[] keyBlob, CngKeyBlobFormat format, CngProvider
 
             SafeNCryptProviderHandle providerHandle = provider.OpenStorageProvider();
             SafeNCryptKeyHandle keyHandle;
-            ErrorCode errorCode = Interop.NCrypt.NCryptImportKey(providerHandle, IntPtr.Zero, format.Format, IntPtr.Zero, out keyHandle, keyBlob, keyBlob.Length, 0);
-            if (errorCode != ErrorCode.ERROR_SUCCESS)
-                throw errorCode.ToCryptographicException();
+            ErrorCode errorCode;
+            
+            if (curve == null)
+            {
+                errorCode = Interop.NCrypt.NCryptImportKey(providerHandle, IntPtr.Zero, format.Format, IntPtr.Zero, out keyHandle, keyBlob, keyBlob.Length, 0);
+                if (errorCode != ErrorCode.ERROR_SUCCESS)
+                {
+                    Exception e = errorCode.ToCryptographicException();
+                    if (errorCode == ErrorCode.NTE_NOT_SUPPORTED &&
+                        (format == CngKeyBlobFormat.EccFullPrivateBlob || format == CngKeyBlobFormat.EccFullPrivateBlob))
```

Comment:
Yes. Actually all 4 are needed. Note that PlatformNotSupported vs. CryptographicException is going to be sometimes ambiguous depending in some cases as we can't always tell if a parameter is not supported or invalid because of platform or because the actual value is wrong

                    if ((errorCode == ErrorCode.NTE_NOT_SUPPORTED || errorCode == ErrorCode.NTE_INVALID_PARAMETER) &&
                        (format == CngKeyBlobFormat.EccFullPrivateBlob || 
                        format == CngKeyBlobFormat.EccFullPrivateBlob ||
                        format == CngKeyBlobFormat.EccFullPublicBlob ||
                        format == CngKeyBlobFormat.EccFullPublicBlob))